roomserver/internal/input/input.go

```diff
@@ -46,6 +47,7 @@ var keyContentFields = map[string]string{
 }
 
 type Inputer struct {
```

Comment:
A block level comment explaining the stream magic would be welcome. Specifically, please have some sort of visual aid e.g:
```
Roomserver input stream

          A BC _   (where each room consumer is up to in the stream)
          v vv v
    ABAABCAABCAA   (letter = subject for each message)
    ---newer--->

 - B and C are all caught up.
 - A is behind and stuck processing an earlier message.
 - _ is the ephemeral consumer which checks the NATS Header for the room ID for the message and
   ensures that a consumer for this room ID has been spun up. See Inputer.Start()
```

Reply:
Assuming my understanding is correct?

Reply:
Yep, that's spot on. I'll adapt it into a block level comment.